File: c/udemy/5-estruturas-repetitivas/loops-indefinidos/media_idades.c
// Faça um programa para ler um número indeterminado de dados, contendo cada um, a idade de um indivíduo. O último dado, que não entrará nos cálculos, contém um valor de idade negativa. Calcular e imprimir a idade média deste grupo de indivíduos. Se for entrado um valor negativo na primeira vez, mostrar a mensagem "IMPOSSIVEL CALCULAR".

#include <stdio.h>

int main() {
    int idade, soma = 0, pessoas = 0;
    double media;

    printf("Idade:\n");
    scanf("%d", &idade);

    if (idade < 0) {
        printf("IMPOSSIVEL CALCULAR");
    } else {
        while (idade >= 0) {
            soma += idade;
            pessoas += 1;

            printf("Idade:\n");
            scanf("%d", &idade);
        }

        if (pessoas > 0) {
            media = (double) soma / pessoas;
            printf("Média: %.2lf anos", media);
        } else {
            printf("Nenhuma idade válida foi inserida.");
        }
    }

    return 0;
}

File: python/udemy/5-estrutura-repetitiva/loops_indefinidos/media_idades.py
# Faça um programa para ler um número indeterminado de dados, contendo cada um, a idade de um indivíduo. O último dado, que não entrará nos cálculos, contém um valor de idade negativa. Calcular e imprimir a idade média deste grupo de indivíduos. Se for entrado um valor negativo na primeira vez, mostrar a mensagem "IMPOSSIVEL CALCULAR"


soma = 0 # acumulador - somatório das idades
cont = 0 # contador - número de pessoas

idade = int(input('Idade: '))

if idade < 0:
    print('IMPOSSÍVEL CALCULAR')

else:
    while idade >= 0:
        soma += idade
        cont += 1
        idade = int(input('Idade: '))

    if cont > 0:
        media = soma / cont
        print(f'Idade média do grupo: {media:.2f}')
